src/spec/controllers/functions_controller_spec.rb

```diff
@@ -48,9 +55,9 @@
     context 'with valid params' do
       it 'creates a new Function' do
         expect do
-          post :create, params: { function: valid_attributes, user_id: user, tags: 'foo, bar, foobar' }
+          post :create, params: { function: valid_attributes.merge(tags_list: tags_list), user_id: user }
```

Comment:
I don't think this should be merged here, the valid_attributes should already include everything, so lets merge it in line 4